src/shared_state/conflicts.md

```diff
@@ -0,0 +1,52 @@
+# Detecting conflicts
+
+Modifications to shared state can be considered conflicting when two conditions are met:
+
+1. They affect an overlapping section of a data structure, such as if two users both try to set the title of the same todo-list item
+
+2. They are [concurrent](https://en.wikipedia.org/wiki/Concurrency_(computer_science)) - meaning that the participants in the system cannot conclusively determine the order in which they occurred
+
+_Concurrent updates which affect unrelated data_ can be resolved without issue, as can _sequential updates to related data_.
+
+Concurrency can effectively be ignored from a client's perspective in schemes that use a central server as a single source of truth.
+In practice one message will always arrive slightly before another, and even if they were truly simultaneous the server could arrange them into an arbitrary order.
+Because this can be done for all messages, all updates can be said to have a [total order](https://en.wikipedia.org/wiki/Total_order), which makes it possible for all clients to compute the same results when they are applied sequentially.
+
+## Conflict resolution in webxdc applications
+
+Many webxdc applications are incapable of producing conflicts, either because they do not use the network communication APIs, or because any given data structure can only have one writer.
+It is worth considering that the webxdc host application might permit a single user to use multiple devices,
+though it is ultimately up to an app's developers whether they want to support such use-cases.
```

Comment:
```suggestion
It is worth considering that the webxdc host application might permit a single user to use multiple devices. For example, a password-manager webxdc app could be used in a "Saved Messages" or "Self-messages" chat so that only a single user's devices will share the application state.  Such a multi-device setting already constitutes a minimal P2P system where total message ordering can not be guaranteed and there is no single source of truth.  
```